bug: Failing "publish docker image" workflow on main

## Context 

There's a failing GitHub action uses the Chaosnet dockerfile. It's been failing several pull requests in a row, as of commit 1921af90c12f4f7444ee38c07659a07155ea6b4b

## Error

<details>
<summary>[Full Error Message]</summary>

```
#27 628.3 gcc: internal compiler error: Segmentation fault signal terminated program cc1
#27 628.3 Please submit a full bug report, with preprocessed source (by using -freport-bug).
#27 628.3 See <file:///usr/share/doc/gcc-12/README.Bugs> for instructions.
#27 673.7 github.com/CosmWasm/wasmvm/internal/api: gcc: signal: segmentation fault (core dumped)
#27 957.3 make: *** [contrib/make/build.mk:135: build] Error 1
#27 ERROR: process "/bin/sh -c make build" did not complete successfully: exit code: 2
------
 > [linux/arm64 builder 6/6] RUN --mount=type=cache,target=/root/.cache/go-build   --mount=type=cache,target=/go/pkg   make build:
589.5 # github.com/linxGnu/grocksdb
589.5 gcc: internal compiler error: Segmentation fault signal terminated program cc1
589.5 Please submit a full bug report, with preprocessed source (by using -freport-bug).
589.5 See <file:///usr/share/doc/gcc-12/README.Bugs> for instructions.
628.3 # github.com/DataDog/zstd
628.3 gcc: internal compiler error: Segmentation fault signal terminated program cc1
628.3 Please submit a full bug report, with preprocessed source (by using -freport-bug).
628.3 See <file:///usr/share/doc/gcc-12/README.Bugs> for instructions.
673.7 github.com/CosmWasm/wasmvm/internal/api: gcc: signal: segmentation fault (core dumped)
957.3 make: *** [contrib/make/build.mk:135: build] Error 1
------
chaosnet.Dockerfile:11
--------------------
  10 |     
  11 | >>> RUN --mount=type=cache,target=/root/.cache/go-build \
  12 | >>>   --mount=type=cache,target=/go/pkg \
  13 | >>>   make build
  14 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c make build" did not complete successfully: exit code: 2
Reference
Check build summary support
Error: buildx failed with: ERROR: failed to solve: process "/bin/sh -c make build" did not complete successfully: exit code: 2
```

</details>

## Workflows

![Image](https://github.com/user-attachments/assets/956ac988-d75c-4b85-a1d2-45e14b2b53ef)

--- a/.github/workflows/chaosnet-docker.yml
+++ b/.github/workflows/chaosnet-docker.yml
@@ -1,4 +1,4 @@
-name: Publish Docker Image
+name: Chaosnet Docker Image
 
 on:
   push:
@@ -9,7 +9,7 @@ on:
 
 jobs:
   chaosnet:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-latest-m
     steps:
       - name: Checkout
         uses: actions/checkout@v4

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -41,6 +41,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [Unreleased]
 
 - [#2104](https://github.com/NibiruChain/nibiru/pull/2074) - chore: update chain IDs
+- [#2206](https://github.com/NibiruChain/nibiru/pull/2206) - ci(chaosnet): fix docker image build
 
 ### Nibiru EVM
 

